### ISSUE TITLE
[Bug 0.5.6] Issues with weapons and attacks

### ISSUE THREAD

**Issue by LeKopic** [kypvalanx/Foundry-VTT-StarWars-SagaEdition#177]
As written in the title, I've tried to add different weapons to a character in a new world, it seems like weapons break actor sheets when I try to equip them.
An error right after trying to equip one.
![image](https://user-images.githubusercontent.com/42682507/166240049-17f4b239-1920-4ebc-b4fe-81579a5b5a67.png)
Then I cannot open the charsheet until I reload Foundry. After reloading the sheet shows weapons as equipped.

When I press the attack button (it rolls an attack without an issue. but there is an error message in the console):
![image](https://user-images.githubusercontent.com/42682507/166240582-c4b44622-ffc2-4f24-995b-9d17ca85e0e3.png)

When I make a stun attack it show almost the same message + there is some issue with the attack chat message:
![image](https://user-images.githubusercontent.com/42682507/166240972-0365c546-d96e-43d4-83f6-7146f0a29815.png)

**Comment by LeKopic** [comment]
As of 0.5.9, a droid with a gun doesn't have any attacks listed and cannot do a full attack
![image](https://user-images.githubusercontent.com/42682507/166268902-d0811035-161e-4398-9a37-d4505dd2cafa.png)

Other species, I think, work as intended

**Comment by LeKopic** [comment]
Hm, the previous bugs, it seems, are now fixed.
But there is a new one (or previously unnoticed) - I can equip added weapons/appendages to some actor only if I open actor sheets in Foundry's Actors Directory, not by opening them through double-clicking on tokens. 
![image](https://user-images.githubusercontent.com/42682507/166447601-dea13b57-5e2c-489c-8b5d-c4ba9f594214.png)

**Comment by kypvalanx** [comment]
i can't seem to reproduce the lates with my dummy droid sheet, can you send me yours?

**Comment by LeKopic** [comment]
Hm, I didn't check it on new actors, but the droids were made on 0.5.3. I can send you character sheets in 4-5 hours, if not later. 
I sent you Dooku's sheet in another issue, maybe it will produce this issue if you switch species to droid and check?

**Comment by kypvalanx** [comment]
ok, i'll give that a try.  i got the bug with dookus sheet figured out.

**Comment by kypvalanx** [comment]
any specific weapon that you were using?

**Comment by LeKopic** [comment]
Nope, I tried different weapons, it didn't matter.

**Comment by LeKopic** [comment]
Hm, ok. I'll send you characters later this day.

**Comment by LeKopic** [comment]
[fvtt-Actor-flame-droid.zip](https://github.com/kypvalanx/Foundry-VTT-StarWars-SagaEdition/files/8611849/fvtt-Actor-flame-droid.zip)
Here is one example.

Here is what happens when I drag weapon from Unequipped to Equipped when opening token's actor sheet by double clicking.
![image](https://user-images.githubusercontent.com/42682507/166490964-8adfe7c3-213f-4f09-8178-7bc8e8495c91.png)

**Comment by kypvalanx** [comment]
I think this issue is resolved, but I'm leaving this issue open until I get affirmation.

**Comment by LeKopic** [comment]
Hmm, no, the issue with opening some character is still there.  Or several issues.
1) I created new Actor -> I added a blaster rifle to the sheet -> I try to equip it -> Issue:
![image](https://user-images.githubusercontent.com/42682507/166555286-17da8eaa-5600-4c70-8de1-63818c16662c.png)
Then I reload Foundry by hitting F5 and now the Blaster Rifle is equipped and works as intended. I can even unequip and equip it as many times as I want to. And now I can add new items without an issue. I am not sure what's the issue, maybe there is a bug with how attacks are created and rendered? Or how attacks use objects. 

2) The second issue, as far as I can tell, only happens when I open some older actors. For example, I have a droideka and I cannot open its sheet.
[fvtt-Actor-droideka.zip](https://github.com/kypvalanx/Foundry-VTT-StarWars-SagaEdition/files/8613883/fvtt-Actor-droideka.zip)
![image](https://user-images.githubusercontent.com/42682507/166556051-7dab29db-c0f5-461f-aab2-de5c7c8bc7fd.png)
And that's a strange error. I suspect it is the same issue with how attacks are created. 

3) Something kinda similar happens with another droid. [fvtt-Actor-t-4-turret-droid.zip](https://github.com/kypvalanx/Foundry-VTT-StarWars-SagaEdition/files/8613967/fvtt-Actor-t-4-turret-droid.zip)
![image](https://user-images.githubusercontent.com/42682507/166557762-d2ecff1e-a39b-44c0-98ad-b653128cd858.png)

**Comment by LeKopic** [comment]
Tested on 0.5.13
Ok, so, I've found out that the issue is reproduceable with new actors. Workflow is as follows - Create an Actor, Set any Species, any Size, Add any weapon -  Try to equip it (ok, after double-checking, you can just create an actor, get any weapon and try to equip it):
![image](https://user-images.githubusercontent.com/42682507/166826266-961a2d2f-4106-44cc-a4ab-261653a1c013.png)

After reloading Foundry the weapon is equipped and is ready to be fired.

But if I am trying to add a weapon that doesn't do damage (tested with gas grenades and grenade launcher) - it break the sheet:
![image](https://user-images.githubusercontent.com/42682507/166826551-7665f8c1-b871-4dc1-896a-7575236b74e1.png)

So I think the system just cannot make an attack if there are no dice to be thrown (as it is the case with T-4 Droid who has a grenade launcher). And Droideka has an amphistaff that probably had set on 1d6/1d6 damage or something. (of course it was the first weapon I saw and dragged into the char sheet several weeks ago)

**Comment by kypvalanx** [comment]
Ah, that explains why I couldn't reproduce it.  I'll take a look

**Comment by LeKopic** [comment]
Ok, there is one last issue relate to this one:
![image](https://user-images.githubusercontent.com/42682507/168401570-b1509df4-e972-45ec-8c41-e12c348167f3.png)
I've add a new weapon (blaster rifle) and then equipped it. It appears the system cannot immediately create an attack, but it works fine after hitting F5.

The error when I try to shoot the "empty" attack:
![image](https://user-images.githubusercontent.com/42682507/168401622-5151dfc5-79cc-453c-8d0d-0b8c0ea2447b.png)

**Comment by kypvalanx** [comment]
well, that's very odd
i'll se what i can find

**Comment by LeKopic** [comment]
Can you replicate this issue on your end?

**Comment by kypvalanx** [comment]
no, but i've added defensive null checks.

**Comment by LeKopic** [comment]
Hm, ok, I'll check it with new checks in ~7-8 hours.
